electrs-blockchain-provider/src/lib.rs

```diff
@@ -304,21 +310,23 @@ impl BlockSource for ElectrsBlockchainProvider {
 }
 
 impl BroadcasterInterface for ElectrsBlockchainProvider {
-    fn broadcast_transaction(&self, tx: &Transaction) {
-        let client = self.client.clone();
-        let host = self.host.clone();
-        let body = bitcoin_test_utils::tx_to_string(tx);
-        std::thread::spawn(move || {
-            match client.post(format!("{host}tx")).body(body).send() {
-                Err(_) => {}
-                Ok(res) => {
-                    if res.error_for_status_ref().is_err() {
-                        // let body = res.text().unwrap_or_default();
-                        // TODO(tibo): log
+    fn broadcast_transactions(&self, txs: &[&Transaction]) {
+        for tx in txs {
+            let client = self.client.clone();
+            let host = self.host.clone();
+            let body = bitcoin_test_utils::tx_to_string(tx);
+            std::thread::spawn(move || {
```

Comment:
It feels wasteful to spawn a thread per transaction, maybe better to move that outside of the for loop?

Reply:
Fixed!